scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/TasksCentricController.java

```diff
@@ -160,28 +162,67 @@ public void selectTask(final Task task) {
             SchedulerServiceAsync scheduler = Scheduler.getSchedulerService();
-            scheduler.getJobInfoDetails(sessionId, jobId, new AsyncCallback<String>() {
-                @Override
-                public void onFailure(Throwable caught) {
-
-                    caught.printStackTrace();
-                    String msg = JSONUtils.getJsonErrorMessage(caught);
-                    LogModel.getInstance().logImportantMessage("Failed to get job info for job " + jobId + ": " + msg);
-                }
-
-                @Override
-                public void onSuccess(String result) {
-                    try {
-                        Job job = SchedulerJSONUtils.getJobInfoFromJson(result);
-                        ((TasksCentricModel) model).setTaskSelectedJob(job);
-                    } catch (JSONException e) {
-                        LogModel.getInstance().logCriticalMessage(e.getMessage());
-                    }
-                }
-            });
+            AsyncCallback<String> callback = new TasksCentricAsyncSelector((TasksCentricModel) model, jobId);
+            scheduler.getJobInfoDetails(sessionId, jobId, callback);
         }
         else{
             ((TasksCentricModel) model).setTaskSelectedJob(null);
         }
     }
+
+    public static class SortSpecifierRestContainer implements Serializable {
+
+        protected List<SortSpecifierRestItem> sortParameters = null;
+
+        public SortSpecifierRestContainer() {
+            sortParameters = new ArrayList<>();
+        }
+
+        public static class SortSpecifierRestItem implements Serializable {
+
+            protected String field;
+            protected String order;
+
+            SortSpecifierRestItem(String field, String order) {
+                this.field = field;
+                this.order = order;
+            }
+
+            public SortSpecifierRestItem() {
+                this.field = "NOTSET";
+                this.order = "ascending";
+            }
+
+            public String toString() {
+                return field + "," + order;
+            }
+        }
+
+        SortSpecifierRestContainer(int size) {
+            sortParameters = new ArrayList<>(size);
+        }
+
+        SortSpecifierRestContainer(String values) {
+            sortParameters = new ArrayList<>();
+            for (String term : values.split(";")) {
+                String[] sortParam = term.split(",");
+                add(sortParam[0], sortParam[1]);
+            }
+        }
+
+        public void add(String field, String order) {
+            sortParameters.add(new SortSpecifierRestItem(field, order));
+        }
+
+        public String toString() {
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0 ; i < sortParameters.size(); i++) {
+                sb.append(sortParameters.get(i).toString());
+                if (i < sortParameters.size() - 1) sb.append(";");
```

Comment:
`sortParameters.size() - 1` could be computed only once and result stored in a variable that is used in the loop. Curly braces around _if_ please :)